pkg/reconciler/instances/ory/hydra/hydra.go

```diff
@@ -0,0 +1,95 @@
+package hydra
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"time"
+)
+
+//go:generate mockery --name=Hydra --outpkg=mock --case=underscore
+//Hydra exposes functionality to trigger hydra specific operations
+type Hydra interface {
+	// TriggerSynchronization triggers the synchronization of OAuth clients between hydra maester and hydra if needed,
+	//that is basically the case when hydra pods started earlier than hydra maester pods, then hydra might be out of sync
+	//as it has potentially lost the OAuth clients in his DB
+	//See also: https://github.com/ory/hydra-maester/tree/master/docs
+	TriggerSynchronization(context context.Context, client kubernetes.Interface, logger *zap.SugaredLogger, namespace string) error
+}
+
+type DefaultHydraClient struct {
+}
+
+// NewDefaultHydraClient returns an instance of DefaultHydraClient
+func NewDefaultHydraClient() *DefaultHydraClient {
+	return &DefaultHydraClient{}
+}
+
+const (
+	hydraPodName           = "app.kubernetes.io/name=hydra"
+	hydraMaesterPodName    = "app.kubernetes.io/name=hydra-maester"
+	hydraMaesterDeployment = "ory-hydra-maester"
+)
+
+func (c *DefaultHydraClient) TriggerSynchronization(context context.Context, client kubernetes.Interface,
+	logger *zap.SugaredLogger, namespace string) error {
+	restartHydraMaesterDeploymentNeeded, err := hydraStartedAfterHydraMaester(context, client, logger, namespace)
```

Comment:
You named the variable exactly how could the function be called. The business use case it to check whether the hydra maester deployment restart is needed. How about renaming the function this way?

Reply:
Indeed, and sure its an option to rename the function exactly like that